tests/e2e/README.md

```diff
@@ -1,4 +1,120 @@
-## Updating the trace format and tests when adjusting the framework
+# End-to-end testing
+
+This folder contains end-to-end tests and related infrastructure.
+The goal of end-to-end tests is to test our application completely.
+End-to-end in this context means tests
+where the whole app (including command line interfaces, interfaces to external components like
+relayers, CometBFT, etc) is in scope for testing.
+
+The high-level overview of what a single test case does is:
+* The test starts a docker container, see [the startup script](testnet-scripts/start-docker.sh)
+* We run a defined sequence of actions and expected states, see as an example the steps for [testing the democracy module](steps_democracy.go)
+    * Actions are things like: submitting transactions to a node, but also things like making nodes double-sign, starting a relayer, starting a new chain, ..., in short anything
+that might modify the system state meaningfully
+    * Expected states defined as parts of the system state that have to match. For example, after an action that modifies validator balances, we would specify
+    the validator balances, so those would be checked against the actual validator balances. We might not care about the governance proposals that are on chain
+    though, and just leave that part of the state unspecified.
+* After each action, the state of the system is queried and compared against the expected state.
+
+## Defining a new test case
+
+This section explains how to define a new test case. For now, let's assume that
+all actions and state checks we want to perform already exist (see (actions.go)[actions.go]
+for possible actions and (state.go)[state.go] for possible state checks).
+Then what we need to do is the following:
+* Create a new test config (or decide on an existing one to reuse), see (config.go)[config.go].
+The test config governs the config parameters of validators and chains that can be run in the test,
+for example we can set genesis parameters using `ChainConfig.GenesisChanges`.
+* Define a sequence of actions and state checks to perform for our test case.
+* Add our test case in the main file (main.go)[main.go].
```

Comment:
```suggestion
* Add the new test case to the main file (main.go)[main.go].
```
Also, add the specific place where to add this new test case. Maybe add some code block or something.

Reply:
Thanks, done!

Reply:
Perfect, done.